docs/integrations/adding-dbos-to-next.md

```diff
@@ -0,0 +1,235 @@
+- Add files containing workflow logic implemented with DBOS. Ensure that the files are being compiled, and load the code from `server.ts` at startup.
+- Add calls to the DBOS code from Next.js code.  This can include server-side page renders, server actions, and route handlers.
+- Ensure that your workflow functions and the DBOS library code do not get bundled into the bundles used by Next.js to handle the requests.  (Failure to do this will lead to a wide variety of error messages.)  This guide covers two techniques: configuring `webpack` to treat the files and modules as external, or using `globalThis` to get a reference to the shared objects and code set up in `server.ts`.
+- Run and troubleshoot the development environment and production build.
+
+# Adding DBOS To Next.JS Applications
+
+## Coding `server.ts`
+`server.ts` (note that, while any file name can be used, `server.ts` is the common convention that is used in this guide) is responsible for initializing DBOS, and launching Next.js.
+
+### Adding DBOS Code To An Existing `server.ts`
+If you already have a `server.ts` or similar file, add the following code to it:
+```typescript
+// ... Other library imports ...
+import { DBOS } from '@dbos-inc/dbos-sdk'; // Import DBOS
+
+//
+// ... In subsequent steps, we will also import any app code that uses DBOS.
+//  This will provide the code and other objects needed for DBOS to recover any workflows...
+
+async function main() { // Adjust if your entrypoint is not named 'main'
+  // ... Configure anything DBOS needs prior to launch ...
+
+  // Launch DBOS after initializing DBOS logic, and before starting 'next'
+  await DBOS.launch();
+
+  // ... Then proceed to start the Next.js server ...
+}
+```
+
+### Adding `server.ts` To A Project
+If your project does not currently have a `server.ts` file or similar, or if you are using the default `server.js` provided by Next.js, the first step is to create one.   This file can be created in the top level of your project, inside `src/`, or in any sensible place within your app structure.
+
+The DBOS demo apps contain example `server.ts` files to copy:
+- [DBOS Next.js Template](https://github.com/dbos-inc/dbos-demo-apps/blob/main/typescript/dbos-nextjs-starter/src/server.ts): This project uses a basic `server.ts` file.
+- [DBOS Task Scheduler](https://github.com/dbos-inc/dbos-demo-apps/blob/main/typescript/nextjs-calendar/src/server.ts): This project uses a more sophisticated `server.ts` file with WebSocket support, etc.
+
+### Compilation Settings For DBOS Code
+Using DBOS with Next.js requires some TypeScript compiler settings that do not match Next.js defaults.  These can either be added to your existing `tsconfig.json`, or if they are incompatible with the existing `tsconfig.json` settings, a separate `tsconfig.server.json` file can be created for use with the server builds.
```

Comment:
This could be softened a bit, right now it sounds really scary ("uncommon settings") when it's really quite mild.